### PR TITLE
add ToJSON and FromJSON instances for Event

### PR DESCRIPTION
**Pull request by shmish111**

### CODE REVIEW

**Comment by commandodev** [comment]
Any idea what the instances look like @shmish111? I suspect the generated ones won't be that usable...

File: src/Network/Monitoring/Riemann/Json.hs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import qualified Data.Text                                  as Text
+import           Network.Monitoring.Riemann.Proto.Attribute as PA
+import           Network.Monitoring.Riemann.Proto.Event     as PE
+import qualified Text.ProtocolBuffers.Header                as P'
+
+instance ToJSON P'.Utf8 where
+  toJSON v = String (Text.pack (P'.uToString v))
+
+instance FromJSON P'.Utf8 where
+  parseJSON (String s) = pure. P'.uFromString . Text.unpack $ s
```
Comment (commandodev) [inline src/Network/Monitoring/Riemann/Json.hs]: You can use `withText` here http://hackage.haskell.org/package/aeson-1.4.0.0/docs/Data-Aeson-Types.html#v:withText

```
parseJSON = withText $ (P'.uFromString . Text.unpack)
```

Should do it

File: src/Network/Monitoring/Riemann/Json.hs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+instance ToJSON PA.Attribute
+
+instance FromJSON PA.Attribute
+
+instance ToJSON PE.Event
+
+instance FromJSON PE.Event where
+  parseJSON = withObject "Event" $ \v -> Event
+    <$> v .: "time"
```
Comment (commandodev) [inline src/Network/Monitoring/Riemann/Json.hs]: Aren't most of these optional?

**Review by commandodev — COMMENTED** [review]
Couple of thought. It might be better to introduce an intermediate type that has the `ToJSON` and `FromJSON` instance and an isomorphism between that and `Event`...

File: src/Network/Monitoring/Riemann/Json.hs (hunk 1 above)
Comment (shmish111) [reply src/Network/Monitoring/Riemann/Json.hs]: Done

File: src/Network/Monitoring/Riemann/Json.hs (hunk 2 above)
Comment (shmish111) [reply src/Network/Monitoring/Riemann/Json.hs]: Fixed

**Comment by shmish111** [comment]
@boothead I've tested with some example json that came from the java client.

**Pull request merged**